pkg/egressgateway/policy.go

```diff
@@ -359,6 +366,28 @@ func ParseCEGP(cegp *v2.CiliumEgressGatewayPolicy) (*PolicyConfig, error) {
 		}
 	}
 
+	// Check for mismatched IP families
```

Comment:
I think this code could be a bit less verbose and can be run earlier in this function. I would propose to add something like the following, right after we have collected all the destination CIDRs:

```suggestion
for _, cidr := range dstCidrList {
	if cidr.Addr().Is6() && policyGwc.egressIP.IsValid() && policyGwc.egressIP.Is4() {
		return nil, fmt.Errorf("mismatched IP families: IPv4 egress IP with IPv6 destination CIDRs")
	}
	if cidr.Addr().Is4() && policyGwc.egressIP.IsValid() && policyGwc.egressIP.Is6() {
		return nil, fmt.Errorf("mismatched IP families: IPv6 egress IP with IPv4 destination CIDRs")
	}
}
```